src/server/protocol.ts

```diff
@@ -2383,8 +2363,9 @@ export interface GeterrRequestArgs {
     /**
      * List of file names for which to compute compiler errors.
      * The files will be checked in list order.
+     * Files with ranges specified will be checked first.

```

Comment:
This says the files with region are checked first but per your updateErrorCheck logic, its checking files in order with delaying full checks when regions are specified after all the files order is done.,  Is this comment uptodate or intentional order  for file1 with region, file2 , file3 region, file4 is to check file1 with region, file3 with region, file2, file 4, file1 semantic and suggestion, file3 semantic and suggestion...

Reply:
The comment is true not because of `updateErrorCheck`, but because of this line in a caller of `updateErrorCheck`: https://github.com/microsoft/TypeScript/pull/57842/files#diff-c9fde3d99427a4b56c409a7dec15c227b21119d0d56bcfab4dbfad23f83118b0R2588.

I did this intentionally, as I think if there are regions of a file visible in the editor, it means they should come before files that are open in the editor but not currently being looked at, if that makes sense.